File: app/ihm/src/domain/Commande.php
<?php

namespace domain;

class Commande
{
    protected $id;
    protected $abonneId;
    protected $dateCommande;
    protected $adresseLivraison;
    protected $dateLivraison;
    protected $lignes;
    protected $prixTotal;

    public function __construct($id, $abonneId, $dateCommande, $adresseLivraison, $dateLivraison, $lignes, $prixTotal)
    {
        $this->id = $id;
        $this->abonneId = $abonneId;
        $this->dateCommande = $dateCommande;
        $this->adresseLivraison = $adresseLivraison;
        $this->dateLivraison = $dateLivraison;
        $this->lignes = $lignes;
        $this->prixTotal = $prixTotal;
    }

    public function getId()
    {
        return $this->id;
    }

    public function getAbonneId()
    {
        return $this->abonneId;
    }

    public function getDateCommande()
    {
        return $this->dateCommande;
    }

    public function getAdresseLivraison()
    {
        return $this->adresseLivraison;
    }

    public function getDateLivraison()
    {
        return $this->dateLivraison;
    }

    public function getLignes()
    {
        return $this->lignes;
    }

    public function getPrixTotal()
    {
        return $this->prixTotal;
    }
}

File: app/commandes/src/main/java/fr/univamu/iut/micromealscommandes/commande/Commande.java
package fr.univamu.iut.micromealscommandes.commande;

import java.util.ArrayList;

/**
 * Entite representant une commande passee par un abonne,
 * contenant les lignes de commande et le prix total
 */
public class Commande {

    protected int id;
    protected int abonneId;
    protected String dateCommande;
    protected String adresseLivraison;
    protected String dateLivraison;
    protected ArrayList<LigneCommande> lignes;
    protected double prixTotal;

    public Commande() {
    }

    /**
     * Cree une commande avec les informations de livraison
     *
     * @param abonneId         identifiant de l'abonné
     * @param adresseLivraison adresse de livraison
     * @param dateLivraison    date souhaitee de livraison
     */
    public Commande(int abonneId, String adresseLivraison, String dateLivraison) {
        this.abonneId = abonneId;
        this.adresseLivraison = adresseLivraison;
        this.dateLivraison = dateLivraison;
        this.lignes = new ArrayList<>();
        this.prixTotal = 0;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getAbonneId() {
        return abonneId;
    }

    public void setAbonneId(int abonneId) {
        this.abonneId = abonneId;
    }

    public String getDateCommande() {
        return dateCommande;
    }

    public void setDateCommande(String dateCommande) {
        this.dateCommande = dateCommande;
    }

    public String getAdresseLivraison() {
        return adresseLivraison;
    }

    public void setAdresseLivraison(String adresseLivraison) {
        this.adresseLivraison = adresseLivraison;
    }

    public String getDateLivraison() {
        return dateLivraison;
    }

    public void setDateLivraison(String dateLivraison) {
        this.dateLivraison = dateLivraison;
    }

    public ArrayList<LigneCommande> getLignes() {
        return lignes;
    }

    public void setLignes(ArrayList<LigneCommande> lignes) {
        this.lignes = lignes;
    }

    public double getPrixTotal() {
        return prixTotal;
    }

    public void setPrixTotal(double prixTotal) {
        this.prixTotal = prixTotal;
    }

    @Override
    public String toString() {
        return "Commande{" + "id=" + id + ", abonneId=" + abonneId + ", dateCommande='" + dateCommande + '\'' + ", adresseLivraison='" + adresseLivraison + '\'' + ", dateLivraison='" + dateLivraison + '\'' + ", lignes=" + lignes + ", prixTotal=" + prixTotal + '}';
    }
}
